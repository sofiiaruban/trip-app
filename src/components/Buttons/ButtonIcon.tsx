import { FC, MouseEventHandler, ReactNode } from 'react'
import { ButtonType } from './constants'
import styles from './Buttons.module.css'

interface ButtonIconProps {
  children: ReactNode
  onClick?: MouseEventHandler<HTMLElement>
  type?: ButtonType
}

const ButtonIcon: FC<ButtonIconProps> = ({
  children,
  onClick,
  type = ButtonType.BUTTON
}) => {
  return (
    <button type={type} onClick={onClick} className={styles.buttonIcon}>
      {children}
    </button>
  )
}

export default ButtonIcon
