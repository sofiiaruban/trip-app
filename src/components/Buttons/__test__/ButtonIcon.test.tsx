import { render } from '@testing-library/react'
import ButtonIcon from '../ButtonIcon'

test('ButtonIcon component matches snapshot', () => {
  const child = <div>"ButtonIcon"</div>

  const { container } = render(<ButtonIcon>{child}</ButtonIcon>)

  expect(container).toMatchSnapshot()
})
