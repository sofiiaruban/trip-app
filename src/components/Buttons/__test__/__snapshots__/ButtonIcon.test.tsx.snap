// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonIcon component matches snapshot 1`] = `
<div>
  <button
    class="buttonIcon"
    type="button"
  >
    <div>
      "ButtonIcon"
    </div>
  </button>
</div>
`;
